docs/src/userguide/limited_api.rst

```diff
@@ -0,0 +1,151 @@
+  complete forward compatibility with future versions of Python isn't assured (and with
+  errors likely to be runtime errors rather than compile-time errors).
+
+Performance
+===========
+
+Running in the Limited API has a notable performance cost.  If this is a concern then you
+should measure it for your own module.  Some rough guidelines follow:
+
+Where the majority of the work involves C-level code the performance loss is likely to be
+low.  This includes code that makes heavy use of typed memoryviews, or code that mainly calls
+an external C library.
+
+Where the majority of the work involves interacting with Python objects the cost is likely to
+be more significant.  As an example, compiling the Cython compiler with the regular C API
+gives a ~35% speed-up compared to not compiling the Cython compiler.  Compiling the Cython
+compiler in the Limited API gives a 0-10% speed-up (depending on the exact version used).
+
+If you are prepared to restrict yourself to Python versions 3.12+, then Cython will use
+the "vectorcall" interface in Limited API mode.  This doesn't enable any new functionality,
+but it does give a noticeable performance improvement. (Outside of the Limited API, Cython
+almost always uses this interface).
+
+Building with the Limited API
+=============================
+
+Cython's usage of the Limited API is controlled by setting the ``Py_LIMITED_API`` macro
+when running the C compiler.  This macro should be set to the version-hex for the
+minimum Python version that you want to support.  Useful version-hexes are:
+
+* ``0x03070000`` - Python 3.7 - the minimum version that Cython supports.
+* ``0x030B0000`` - Python 3.11 - the first version to support typed memoryviews.
+* ``0x030C0000`` - Python 3.12 - the first version to support vectorcall (performance
+  improvement).
+  
+As well as setting the ``Py_LIMITED_API`` macro, you should also name the compiled
+extension modules to indicate their use of the Stable ABI.  On OS X and Linux, this
+means the extension modules names should end with ``.abi3.so``.
+
+A number of examples are shown below for different build systems, but the
+same basic principles apply to any other build system.
+
+Setuptools and setup.py
+-----------------------
+
+Using setup.py to control the compilation (as shown in the main :ref:`compilation`
+documentation)::
+
+    from setuptools import Extension, setup
+    from Cython.Build import cythonize
+    
+    setup(
+        ext_modules=cythonize([
+            Extension(
+                name="cy_code",
+                sources=["cy_code.pyx"],
+                define_macros=[
+                    ("Py_LIMITED_API", 0x03070000),
+                ],
+                py_limited_api=True
+            ),
+        ]))
+        
+The key differences are two arguments to ``Extension``:  ``define_macros`` and ``py_limited_api``.
+The ``py_limited_api`` argument controls the naming of the extension module.
+
+Scikit-build
```

Comment:
We haven't previously documented some of the alternative build systems like Scikit-build and Meson. I think we probably should at some point (although I'm definitely not an expert in either). I already had some examples prepared for these, so it seemed a good place to start.